_posts/2019-06-19-load-image-data.md

```diff
@@ -0,0 +1,1095 @@
+  <polygon points="0.000000,0.000000 41.887587,0.000000 41.887587,25.412617 0.000000,25.412617" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="20.943793" y="45.412617" font-size="1.0rem" font-weight="100" text-anchor="middle" >199</text>
+  <text x="61.887587" y="12.706308" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(0,61.887587,12.706308)">3</text>
+
+
+  <!-- Horizontal lines -->
+  <line x1="111" y1="0" x2="135" y2="24" style="stroke-width:2" />
+  <line x1="111" y1="120" x2="135" y2="144" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="111" y1="0" x2="111" y2="120" style="stroke-width:2" />
+  <line x1="135" y1="24" x2="135" y2="144" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="111.000000,0.000000 135.664918,24.664918 135.664918,144.664918 111.000000,120.000000" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="111" y1="0" x2="201" y2="0" style="stroke-width:2" />
+  <line x1="135" y1="24" x2="225" y2="24" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="111" y1="0" x2="135" y2="24" style="stroke-width:2" />
+  <line x1="201" y1="0" x2="225" y2="24" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="111.000000,0.000000 201.000000,0.000000 225.664918,24.664918 135.664918,24.664918" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Horizontal lines -->
+  <line x1="135" y1="24" x2="225" y2="24" style="stroke-width:2" />
+  <line x1="135" y1="144" x2="225" y2="144" style="stroke-width:2" />
+
+  <!-- Vertical lines -->
+  <line x1="135" y1="24" x2="135" y2="144" style="stroke-width:2" />
+  <line x1="225" y1="24" x2="225" y2="144" style="stroke-width:2" />
+
+  <!-- Colored Rectangle -->
+  <polygon points="135.664918,24.664918 225.664918,24.664918 225.664918,144.664918 135.664918,144.664918" style="fill:#ECB172A0;stroke-width:0"/>
+
+  <!-- Text -->
+  <text x="180.664918" y="164.664918" font-size="1.0rem" font-weight="100" text-anchor="middle" >768</text>
+  <text x="245.664918" y="84.664918" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(-90,245.664918,84.664918)">1024</text>
+  <text x="113.332459" y="152.332459" font-size="1.0rem" font-weight="100" text-anchor="middle" transform="rotate(45,113.332459,152.332459)">201</text>
+</svg>
+</td>
+</tr>
+</table>
+
+That's a 180 GB logical array, composed of around 600 chunks, each of size 300
+MB.  We can now do normal NumPy like computations on this array using [Dask
+Array](https://docs.dask.org/en/latest/array.html), but we'll save that for a
+future post.
+
+
+Saving Data
+-----------
+
+To simplify data loading in the future, we store this in a large chunked
+array format like [Zarr]( https://zarr.readthedocs.io/ ) using the [to_zarr](
+https://docs.dask.org/en/latest/array-api.html#dask.array.Array.to_zarr )
+method.
+
+```python
+a.to_zarr("mydata.zarr")
```

Comment:
@jakirkham I am sure you have already thought of this, but it would be wonderful to preserve the [image metadata](https://github.com/imageio/imageio/issues/382) in attributes when saving to zarr, i.e. the spatial origin and spacing.

Reply:
Thanks @thewtex. Added a small note about leveraging attributes to store metadata. 🙂